go-controller/pkg/ovn/egressip.go

```diff
@@ -1612,13 +1647,23 @@ func (oc *DefaultNetworkController) generateCacheForEgressIP() (map[string]egres
 	if err != nil {
 		return nil, err
 	}
+	cloudPrivateIPConfigMap, err := oc.getCloudPrivateIPConfigMap()
+	if err != nil {
+		return nil, err
+	}
 	for _, egressIP := range egressIPs {
 		egressIPCache[egressIP.Name] = egressIPCacheEntry{
 			egressPods:       make(map[string]sets.Set[string]),
 			gatewayRouterIPs: sets.New[string](),
 			egressIPs:        map[string]string{},
 		}
 		for _, status := range egressIP.Status.Items {
+			if util.PlatformTypeIsEgressIPCloudProvider() {
+				cloudPrivateIPConfigName := ipStringToCloudPrivateIPConfigName(status.EgressIP)
```

Comment:
add a comment here, explaining that cache must not include eips that cncc does not know about. Is this really needed? I see `oc.getCloudPrivateIPConfigMap()` being called in `syncEgressIPsStatus()` so I'm a bit confused here.

Reply:
yeah we should wait for IC to land. We are moving the cloud private IP config watcher to the cluster-manager now. So we should add a new sync function for this in cluster-manager. Master will not be watching private IPs anymore.